src/config.rs

```diff
@@ -0,0 +1,418 @@
+        CF_DEFAULT => (0.25, 0, usize::MAX),
+        CF_LOCK => (0.02, LOCKCF_MIN_MEM, LOCKCF_MAX_MEM),
+        CF_WRITE => (0.15, 0, usize::MAX),
+        CF_RAFT => (0.02, RAFTCF_MIN_MEM, RAFTCF_MAX_MEM),
+        _ => unreachable!(),
+    };
+    let mut size = (total_mem as f64 * radio) as usize;
+    if size < min {
+        size = min;
+    } else if size > max {
+        size = max;
+    }
+    size / MB as usize
+}
+
+#[derive(Clone, Serialize, Deserialize)]
+#[serde(default)]
+#[serde(rename_all = "kebab-case")]
+pub struct CfConfig {
+    #[serde(skip)]
+    pub cf: String,
+    pub block_size: ReadableSize,
+    pub block_cache_size: ReadableSize,
+    pub cache_index_and_filter_blocks: bool,
+    pub use_bloom_filter: bool,
+    pub whole_key_filtering: bool,
+    pub bloom_filter_bits_per_key: i32,
+    pub block_based_bloom_filter: bool,
+    #[serde(with = "compression_type_level_serde")]
+    pub compression_per_level: [DBCompressionType; 7],
+    pub write_buffer_size: ReadableSize,
+    pub max_write_buffer_number: i32,
+    pub min_write_buffer_number_to_merge: i32,
+    pub max_bytes_for_level_base: ReadableSize,
+    pub target_file_size_base: ReadableSize,
+    pub level0_file_num_compaction_trigger: i32,
+    pub level0_slowdown_writes_trigger: i32,
+    pub level0_stop_writes_trigger: i32,
+    pub max_compaction_bytes: ReadableSize,
+    #[serde(with = "config::compaction_pri_serde")]
+    pub compaction_pri: CompactionPriority,
+}
+
+impl Default for CfConfig {
+    fn default() -> CfConfig {
+        CfConfig {
+            cf: CF_DEFAULT.to_owned(),
+            block_size: ReadableSize::kb(64),
+            block_cache_size: ReadableSize::mb(memory_for_cf(CF_DEFAULT) as u64),
+            cache_index_and_filter_blocks: true,
+            use_bloom_filter: true,
+            whole_key_filtering: true,
+            bloom_filter_bits_per_key: 10,
+            block_based_bloom_filter: false,
+            compression_per_level: [DBCompressionType::No,
+                                    DBCompressionType::No,
+                                    DBCompressionType::Lz4,
+                                    DBCompressionType::Lz4,
+                                    DBCompressionType::Lz4,
+                                    DBCompressionType::Zstd,
+                                    DBCompressionType::Zstd],
+            write_buffer_size: ReadableSize::mb(128),
+            max_write_buffer_number: 5,
+            min_write_buffer_number_to_merge: 1,
+            max_bytes_for_level_base: ReadableSize::mb(512),
+            target_file_size_base: ReadableSize::mb(32),
+            level0_file_num_compaction_trigger: 4,
+            level0_slowdown_writes_trigger: 20,
+            level0_stop_writes_trigger: 36,
+            max_compaction_bytes: ReadableSize::gb(2),
+            compaction_pri: CompactionPriority::MinOverlappingRatio,
+        }
+    }
+}
+
+impl CfConfig {
+    fn write_config() -> CfConfig {
```

Comment:
`toml::from_str` will convert content of tikv.toml to `TiKvConfig` automatically, which contains `CfConfig`.